lib/dal/src/workspace_snapshot/graph.rs

```diff
@@ -1600,6 +1606,36 @@ impl WorkspaceSnapshotGraph {
         algo::has_path_connecting(&self.graph, self.root_index, node, None)
     }
 
+    /// Import a single node from the `other` [`WorkspaceSnapshotGraph`] into `self`, if it does
+    /// not already exist in `self`.
+    pub fn find_in_self_or_import_node_from_other(
```

Comment:
In the `Update::NewEdge` case, we call `find_in_self_or_create_using_onto` which calls `import_subgraph` even if the node exists in both graphs. I'm wondering if we should use this new method in the case that we find an equivalent node in other?

Reply:
So, walking things through:
* `detect_confilcts_and_updates` walks the `onto` graph as its basis for detecting things.
* If we don't find any candidates for equivalency for a particular node in `onto` from the nodes in `to_rebase`, then we do nothing, and prune the walk of the `onto` graph at that point.
* When processing a container in `find_container_membership_conflicts_and_updates` (processing a/the parent of the potentially new node).
* There is an entry in `only_onto_edges` for the edge pointing to a node. (May or may not be to a new node)
    * We generate an `Update::NewEdge`
* If the node exists already in `to_rebase` (not new)
  * When processing the DFS event for the node, we continue walking the `onto` graph through this node as long as we continue to find equivalent nodes in `to_rebase`, generating `Update`s as necessary.
* If the node does _not_ already exist in `to_rebase` (it's new)
  *  When processing the DFS event for the node, we prune the `onto` graph walk as we could not find an equivalent node, and do not generate any `Update` for the node that only exists in `onto`.

I that when we `Update:NewEdge` in `perform_updates` that it should be a no-op if we find an equivalent node in`to_rebase` as far as importing/updating nodes goes, and that if there is no equivalent node in `to_rebase` then we want to import the "new" bits of the subgraph.

So, I don't think the new `find_in_self_or_import_node_from_other` is quite right there as it won't import the rest of the new subgraph (only the root of it), but I think you're correct that `find_in_self_or_create_using_onto` isn't quite right either as it still tries to import the entire sub-graph from `onto` in the case where it did find an equivalent pre-existing node in`to_rebase`.